### PR TITLE
[feat]: [CDS-86071]: Add delete event for github trigger

### PR DESCRIPTION
**Pull request by rathodmeetsatish**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56854)

### CODE REVIEW

**Comment by rathodmeetsatish** [comment]
trigger uts

**Comment by rathodmeetsatish** [comment]
trigger uts

File: pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/beans/source/webhook/v2/github/event/GithubDeleteSpec.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+@JsonIgnoreProperties(ignoreUnknown = true)
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@OwnedBy(PIPELINE)
+public class GithubDeleteSpec implements GithubEventSpec {
+  String connectorRef;
+  String repoName;
+  List<TriggerEventDataCondition> headerConditions;
+  List<TriggerEventDataCondition> payloadConditions;
+  String jexlCondition;
+  boolean autoAbortPreviousExecutions;
```
Comment (sribalij) [inline pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/beans/source/webhook/v2/github/event/GithubDeleteSpec.java:38]: Is autoAbort a valid use case?

**Review by sribalij — COMMENTED** [review]
Is the event supported on go-scm side?

**Comment by rathodmeetsatish** [comment]
> Is the event supported on go-scm side?

@sribalij delete event is parsed as branch or tag eventso on BE when we recieve branch or tag event with delete action we consider it as delete event

**Comment by rathodmeetsatish** [comment]
trigger uts

File: pipeline-service/service/src/main/java/io/harness/pms/triggers/webhook/helpers/TriggerEventExecutionHelper.java [hunk 2]
```diff
@@ -281,6 +282,12 @@ TriggerPayload getTriggerPayloadForWebhookTrigger(ParseWebhookResponse parseWebh
       if (parseWebhookResponse.hasRelease()) {
         builder.setParsedPayload(ParsedPayload.newBuilder().setRelease(parseWebhookResponse.getRelease()).build())
             .build();
+      } else if (parseWebhookResponse.hasBranch()
```
Comment (vcalasansh) [inline pipeline-service/service/src/main/java/io/harness/pms/triggers/webhook/helpers/TriggerEventExecutionHelper.java:285]: Are we sure this is working with CI pipelines, after the changes in this file?

**Review by vcalasansh — APPROVED** [review]
:lgtm:

> Reviewed 6 of 9 files at r1, 1 of 1 files at r2, 8 of 8 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56854#-:-NnBPoNyAdun8Hh3WKWU:bnfp4nl)* status: 15 file(s) to review. 1 of 2 approvals obtained.

**Review by sribalij — APPROVED** [review]

**Comment by sribalij** [comment]
trigger ut2, ut4

File: pipeline-service/service/src/main/java/io/harness/pms/triggers/webhook/helpers/TriggerEventExecutionHelper.java (hunk 2 above)
Comment (rathodmeetsatish) [reply pipeline-service/service/src/main/java/io/harness/pms/triggers/webhook/helpers/TriggerEventExecutionHelper.java:285]: yes

**Comment by rathodmeetsatish** [comment]
trigger messagecheck,review

**Pull request merged**